File: 2019/day01/python/main.py
def part1(input):
    return sum(int(i) // 3 - 2 for i in input)

def part2(input):
    fuel = 0

    for i in [int(num) for num in input]:
        while (i := i // 3 - 2) > 0:
            fuel += i

    return fuel

input = open('../input.txt').read().splitlines()

print(part1(input))
print(part2(input))

File: 2019/day01/rust/src/main.rs
use std::fs;

fn main() {
    let input: Vec<u32> = fs::read_to_string("../input.txt")
        .unwrap()
        .lines()
        .map(|line| line.parse().unwrap())
        .collect();

    let mut total_fuel = (0, 0);

    for val in input {
        let mut fuel = val / 3 - 2;

        total_fuel.0 += fuel;

        loop {
            total_fuel.1 += fuel;

            match fuel.checked_div(3).and_then(|res| res.checked_sub(2)) {
                Some(value) => fuel = value,
                None => break,
            }
        }
    }

    println!("{}\n{}", total_fuel.0, total_fuel.1);
}
